### PR TITLE
[14.0][IMP] fieldservice_maintenance: maintenance request is created …

### PR DESCRIPTION
**Pull request by PicchiSeba**
…even if location_id is not set. But fsm_order is not generated.

Hi,
this PR references issue #1025.

This is the solution that i propose to fix the situation that is affecting me.
I believe it could be useful to others as well.

EDIT: spelling

### CODE REVIEW

**Comment by PicchiSeba** [comment]
I just rebased the relative commit on the most recent 14.0 version

**Comment by hparfr** [comment]
What do you think of having a flag `create fsm.order` on maintenance requests ? 
Because, you may not want fsm.order at all even if there is a location attached.

**Comment by PicchiSeba** [comment]
I will look into it and let you know

File: fieldservice_maintenance/models/maintenance_request.py [hunk 1]
```diff
@@ -24,23 +28,26 @@ def create(self, vals):
...
-                }
-            )
-
-            request.fsm_order_id = fsm_order_id
+            if request.create_fsm_order:
+                if not fsm_equipment.current_location_id.id:
+                    _logger.info(
+                        _(
+                            "Missing location on fsm_equipment %s. fsm_order was not created"
+                            % fsm_equipment.id
```
Comment (max3903) [inline fieldservice_maintenance/models/maintenance_request.py]: @PicchiSeba What do you think of adding this message to the chatter of the maintenance request as well? The user does not see the logs.

Comment (PicchiSeba) [reply fieldservice_maintenance/models/maintenance_request.py]: makes sense. thank you

Comment (PicchiSeba) [reply fieldservice_maintenance/models/maintenance_request.py]: Hi, sorry for being late. I applied the suggestion

**Review by max3903 — APPROVED** [review]

**Comment by PicchiSeba** [comment]
I just updated the relative test file

**Comment by PicchiSeba** [comment]
I'm sorry for the many force pushes, but I don't understand what's the matter with codecov.

The pipeline error says that i decreased code coverage by 0.01% but checking on codecov's website I see 100% coverage.
https://app.codecov.io/gh/OCA/field-service/commit/5f90535da3a67d4ffa82a00ec12bb23073bb35d1
Am I reading that correctly or am I missing something?

**Pull request merged**